addons/analytic/models/analytic_mixin.py

```diff
@@ -54,31 +60,60 @@ def _condition_to_sql(self, alias: str, fname: str, operator: str, value, query:
             ))
             operator = 'in' if operator in ('=', 'ilike') else 'not in'
 
-        analytic_distribution_sql = self._field_to_sql(alias, 'analytic_distribution', query)
+        analytic_accounts_query = self._query_analytic_accounts()
         value = [str(id_) for id_ in value if id_]  # list of ids -> list of string
-        if operator == 'in':  # 'in' -> ?|
+        if operator == 'in':  # keys can be comma-separated ids, we will fetch each id separately
             return SQL(
-                "%s ?| ARRAY[%s]",
-                analytic_distribution_sql,
+                "%s && %s",
+                analytic_accounts_query,
                 value,
             )
         if operator == 'not in':
             return SQL(
-                "(NOT %s ?| ARRAY[%s] OR %s IS NULL)",
-                analytic_distribution_sql,
+                "(NOT %s && %s OR %s IS NULL)",
+                analytic_accounts_query,
                 value,
-                analytic_distribution_sql,
+                self._field_to_sql(alias, 'analytic_distribution', query),
             )
         raise UserError(_('Operation not supported'))
 
     def _read_group_groupby(self, groupby_spec: str, query: Query) -> SQL:
+        """To group by `analytic_distribution`, we first need to separate the analytic_ids and associate them with the ids to be counted
+        Do note that only '__count' can be passed in the `aggregates`"""
         if groupby_spec == 'analytic_distribution':
-            return SQL(
-                'jsonb_object_keys(%s)',
-                self._field_to_sql(self._table, 'analytic_distribution', query),
-            )
+            query._tables = {
+                'distribution': SQL(
+                    r"""(SELECT DISTINCT %s, (regexp_matches(jsonb_object_keys(%s), '\d+', 'g'))[1]::int AS account_id FROM %s WHERE %s)""",
+                    self._get_count_id(query),
+                    self._field_to_sql(self._table, 'analytic_distribution', query),
+                    query.from_clause,
+                    query.where_clause,
+                )
+            }
+
+            # After using the from and where clauses in the nested query, they are no longer needed in the main one
+            query._joins = {}
+            query._where_clauses = []
+            return SQL("account_id")
+
         return super()._read_group_groupby(groupby_spec, query)
 
+    def _read_group_select(self, aggregate_spec: str, query: Query) -> SQL:
+        if '(regexp_matches(jsonb_object_keys(' in query.from_clause.code and aggregate_spec != '__count':
```

Comment:
I would say that you could base on the table name that you defined in `read_group_groupby` instead of the the string search